2024-06-10/server/main.py

```diff
@@ -0,0 +1,161 @@
+"""
+Coding challenge -- playing with APIs.
+I built this half, and Spencer wrote functions to communicate
+to the various endpoints on this API.
+- SAM 2024-06-10
+"""
+
+import json
+import os
+import inspect
+import uuid
+
+from fastapi import APIRouter, FastAPI
+
+app = FastAPI()
+router = APIRouter(prefix="/api")
+
+USERS = "users.txt"
+
+
+def read_users() -> dict:
+    with open(USERS, "r") as file_handle:
+        try:
+            users = json.load(file_handle)
+        except Exception as exc:
+            print(f"FAILED TO LOAD USERS: {exc}")
+            users = {}
+    return users
+
+
+@router.post("/users/add")
+def add_user(
+    name: str,
+    email: str | None = None,
+    phone: str | None = None,
+    address: str | None = None,
+) -> int:
+    users = read_users()
+    with open(USERS, "w") as file_handle:
```

Comment:
Does this write a new file every time we call this method?

Reply:
That's right. We're going dumb here, where we just read the whole file in, and overwrite it when we're done.